Rubberduck.CodeAnalysis/Inspections/Concrete/FunctionReturnValueNotUsedInspection.cs

```diff
@@ -46,159 +44,54 @@ namespace Rubberduck.Inspections.Concrete
     /// End Function
     /// ]]>
     /// </example>
-    public sealed class FunctionReturnValueNotUsedInspection : InspectionBase
+    public sealed class FunctionReturnValueNotUsedInspection : IdentifierReferenceInspectionBase
```

Comment:
Keeping in mind that renaming an inspection is a breaking change as far as user configurations go, should this one be renamed `DiscardedFunctionReturnValue` since we're introducing `FunctionReturnValueNeverUsed`? That would do away with the "NotUsed"/"NeverUsed" dichotomy.

Reply:
IIRC we have that dichotomy in other inspection names as well (didn't check though). If that is the case, then we should either "fix" all of them or keep all of them the same (to keep breaking change occurences minimal)

Reply:
My problem with the old name is that it led to confusion more than once. It suggests that it is about the usage. However, the original inspection is about the declaration.

Reply:
I just had a look: we only use _NotUsed_, so far. However, in all other cases, it is not ambiguous. (Here, the problem is that the return value can mean the individual value.) 

I will rename `FunctionReturnValueNotUsedInspection` to `FunctionReturnValueDiscardedInspection`.

Should I rename `FunctionReturnValueNeverUsedInspection` to `FunctionReturnValueAlwaysDiscardedInspection`, as well?

Reply:
Hmm. I like how "AlwaysDiscarded" avoids "Not"/"Never" wording. Might as well!